### PR TITLE
boards: arm: add new rev. for STM32F3Disco board

### PR DESCRIPTION
**Pull request by helo9**
New STM32F3 Discovery boards are shipped with different
sensors. This PR adds a new revision for the board, supporting
the new acceloremeter/magnetometer sensor.

### CODE REVIEW

File: boards/arm/stm32f3_disco/doc/index.rst [hunk 1]
```diff
@@ -28,9 +28,11 @@ started quickly. Here are some highlights of the STM32F3DISCOVERY board:
 
 - Two push-buttons: USER and RESET
 - USB USER with Mini-B connector
-- L3GD20, ST MEMS motion sensor, 3-axis digital output gyroscope
+- L3GD20, ST MEMS motion sensor, 3-axis digital output gyroscope; PCB version
+  E uses I3G4250D instead
 - LSM303DLHC, ST MEMS system-in-package featuring a 3D digital linear
-  acceleration sensor and a 3D digital magnetic sensor
+  acceleration sensor and a 3D digital magnetic sensor; for PCB
+  version E LSM303AGR is mounted instead
```
Comment (erwango) [inline boards/arm/stm32f3_disco/doc/index.rst:35]: This part could be better formatted

File: boards/arm/stm32f3_disco/revision.cmake [hunk 2]
```diff
@@ -0,0 +1,4 @@
+board_check_revision(
+        FORMAT LETTER
+        DEFAULT_REVISION E
```
Comment (erwango) [inline boards/arm/stm32f3_disco/revision.cmake:3]: Please don't change default revision.
Till now people used older revision and may have scripts using it.
Setting the new revision as default is not backward compatible.

File: boards/arm/stm32f3_disco/doc/index.rst [hunk 3]
```diff
@@ -210,9 +212,12 @@ Here is an example for the :ref:`hello_world` application.
 
 .. zephyr-app-commands::
    :zephyr-app: samples/hello_world
-   :board: stm32f3_disco
+   :board: stm32f3_disco@E
    :goals: build flash
```
Comment (erwango) [inline boards/arm/stm32f3_disco/doc/index.rst:216]: Please let this part as is (default `stm32f3_disco` remains available) and add a new paragraph to introduce the new E revision

File: boards/arm/stm32f3_disco/doc/index.rst [hunk 4]
```diff
@@ -239,7 +244,7 @@ You can debug an application in the usual way.  Here is an example for the
 
 .. zephyr-app-commands::
    :zephyr-app: samples/hello_world
-   :board: stm32f3_disco
+   :board: stm32f3_disco@E
```
Comment (erwango) [inline boards/arm/stm32f3_disco/doc/index.rst:247]: Ditto

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for adding this!

Some comments around fact that existing should remain default for backward compatibility.
One question: is that possible to use lower cap for revision ? (@tejlmand maybe?)

**Review by erwango — APPROVED** [review]
Thanks for the changes!

**Comment by tejlmand** [comment]
> One question: is that possible to use lower cap for revision ? (@tejlmand maybe?)

Not when using `board_check_revision()`, you would need to do a custom board revision handler.
@erwango Feel free to open an enhancement issue if you think this should be supported.

**Comment by erwango** [comment]
> > One question: is that possible to use lower cap for revision ? (@tejlmand maybe?)
> 
> Not when using `board_check_revision()`, you would need to do a custom board revision handler.
> @erwango Feel free to open an enhancement issue if you think this should be supported.

Thanks for feedback, I've just submitted https://github.com/zephyrproject-rtos/zephyr/issues/33921

**Review by FRASTM — APPROVED** [review]
with change in the commit name : 'boar**d**s: arm: add new rev. for STM32F3Dis**c**o board'

**Review by galak — CHANGES_REQUESTED** [review]
Can you fix the commit message to be `boards: ...`

**Comment by helo9** [comment]
> Can you fix the commit message to be `boards: ...`



> Can you fix the commit message to be `boards: ...`

:blush: 

sure

**Review by galak — APPROVED** [review]

**Pull request merged**